api/service.go

```diff
@@ -93,21 +319,108 @@ func (s *service) Listen(req *pb.ListenRequest, server pb.API_ListenServer) erro
 
 	listener := store.StateChangeListen()
 	defer listener.Discard()
-	for range listener.Channel() {
-		err := model.ReadTxn(func(txn *es.Txn) error {
-			var res string
-			if err := txn.FindByID(core.EntityID(req.EntityID), &res); err != nil {
-				return err
+
+	for {
+		select {
+		case _ = <-server.Context().Done():
+			return nil
```

Comment:
If the context is canceled (because the client canceled the request), we need to return